### PR TITLE
Bug fix for unpredictable Exodus output behavior in --mesh-only mode

### PR DESCRIPTION
**Pull request by shikhar413**
## Impact

Addresses #24496

### CODE REVIEW

**Comment by shikhar413** [comment]
@eshemon FYI

File: framework/src/actions/MeshOnlyAction.C [hunk 1]
```diff
@@ -110,26 +110,32 @@ MeshOnlyAction::act()
...
           if (restrict_element_id_names)
+          {
             element_id_names = params.get<std::vector<std::string>>("extra_element_ids_to_output");
+            // Check which element id names actually are defined on the mesh, remove from
+            // element_id_names if they don't belong
+            for (auto it = element_id_names.begin(); it != element_id_names.end();)
+            {
+              // Erase contents of iterator and return iterator if element integer does not exist
+              if (!output_mesh.has_elem_integer(*it))
+                it = element_id_names.erase(it);
```
Comment (GiudGiud) [inline framework/src/actions/MeshOnlyAction.C:121]: should we have at least a warning here? we re disobeying a user input

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests [hunk 2]
```diff
@@ -158,6 +158,43 @@
     exodiff = 'core_metadata_transfer.e'
     recover = false
   []
+  [hex_metadata_transfer_mesh_only]
+    requirement = 'The system shall generate a 2D hexagonal core mesh that transfers metadata correctly across RGMB mesh generators in mesh only with extra element ids that have been defined with default values for cetain id names'
```
Comment (GiudGiud) [inline modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests:162]:
```suggestion
    requirement = 'The system shall generate a 2D hexagonal core mesh that transfers metadata correctly across RGMB mesh generators in mesh only with extra element ids that have been defined with default values for certain id names'
```

File: framework/src/actions/MeshOnlyAction.C [hunk 3]
```diff
@@ -144,7 +150,7 @@ MeshOnlyAction::act()
           eeid_vars.push_back(output_mesh.get_elem_integer_name(i));
           for (const auto & elem : output_mesh.element_ptr_range())
           {
-            eeid_soln[soln_index] = elem->get_extra_integer(i);
+            eeid_soln[soln_index] = (int)elem->get_extra_integer(i);
```
Comment (GiudGiud) [inline framework/src/actions/MeshOnlyAction.C:153]: why do we need this

**Review by GiudGiud — COMMENTED** [review]
looks good
minor things before this gets in

File: framework/src/actions/MeshOnlyAction.C (hunk 1 above)
Comment (shikhar413) [reply framework/src/actions/MeshOnlyAction.C:122]: Added a warning here

**Review by GiudGiud — DISMISSED** [review]

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the patch!

**Pull request merged**